x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -515,6 +517,7 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
         useLegacyAlerts: true,
         isSpaceAware: true,
         secondaryAlias: config.signalsIndex,
+        formatAlert: formatAlertForNotificationActions as unknown as FormatAlert<never>,
```

Comment:
`AlertData` should be passed as a type when registering a ruleType to avoid this type casting.
I tried but couldn't manage to to figure out the `AlertData` type for the security rule types.
It also causes too much changes in ruleRegistry as well.

We can create a follow-on issue to pass the AlertData to ruleRegistry -> actionsClient,
or wait for all the ruleTypes to adopt the new AlertsClient and remove ruleRegistry.